lib/playlistGenerator/PlaylistGenerator.js

```diff
@@ -413,53 +423,26 @@ PlaylistGenerator.prototype.toJSON = function () {
     return diag;
 };
 
-var checkTranscodingProfileChange = function () {
-    var that = this;
-    var entryPath = persistenceFormat.getEntryBasePath(that.entryId);
-    var reg = new RegExp("(?:" + that.entryId + "\\/)([0-9]*)\\b", "i");
-    return qio.listDirectoryTree(entryPath)
-        .then(function(result) {
-            return _.chain(result)
-                .filter(function(d) {
-                    return d.match(reg) !== null;
-                })
-                .map(function(d) {
-                    return d.match(reg)[1];
-                })
-                .uniq(function(f) {
-                    return f;
-                })
-                .value();
-        })
-        .then(function(result) {
-            var playingFlavorsString = _.map(that.flavorsObjArray, function(f) {
-                return f.name;
-            }).sort().join();
-            if (result.sort().join() !== playingFlavorsString) {
-                that.logger.info("Transcoding profile changed, cleaning entry folder");
-                return fsUtils.cleanFolder(entryPath);
-            }
-        })
-        .catch(function(error) {
-            that.logger.error("Error cleaning folder because of transcoding profile change: %s", ErrorUtils.error2string(error));
-        });
+PlaylistGenerator.prototype.checkTranscodingProfileChange = function() {
+    let that = this;
+    let playedFlavors = _.map(that.playlistImp.inner.sequences, (seq) => { return seq.inner.id }).sort().join();
```

Comment:
Because the function that's calling it is expecting a promise to determine how to move forward